### DISCUSSION TITLE
property: Optional[ComplexObject] = field(None,  nullable=True) doesn't create a nullable field in openapi spec

### DISCUSSION THREAD

**Discussion by zhukovgreen** [fastapi/fastapi#7106] [Questions] [answered]
### First check

* [X] I added a very descriptive title to this issue.
* [X] I used the GitHub search to find a similar issue and didn't find it.
* [X] I searched the FastAPI documentation, with the integrated search.
* [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [X] I already read and followed all the tutorial in the docs and didn't find an answer.
* [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [X] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
import enum

from typing import Optional

import pytest

from fastapi import FastAPI
from pydantic import BaseModel, Field


app = FastAPI()


class RHELVersions(enum.Enum):
    """Available RHEL versions within the database."""

    el7: str = "RHEL 7"
    el8: str = "RHEL 8"
    el9: str = "RHEL 9"


class DeviceEntry(BaseModel):
    """A model which represents the single hardware device."""

    simple_property: Optional[bool] = Field(None, nullable=True)
    complex_property: Optional[RHELVersions] = Field(None, nullable=True)


@app.get("/", response_model=DeviceEntry)
def read_root() -> DeviceEntry:
    return DeviceEntry(simple_property=None, complex_property=None)


def test_case():
    spec = app.openapi()
    assert spec["components"]["schemas"]["DeviceEntry"]["properties"][
        "simple_property"
    ]["nullable"]
    with pytest.raises(KeyError):
        assert spec["components"]["schemas"]["DeviceEntry"]["properties"][
            "complex_property"
        ]["nullable"]


"""
  +  'components': {'schemas': {'DeviceEntry': {'description': 'A model which '
  +                                                            'represents the '
  +                                                            'single hardware '
  +                                                            'device.',
  +                                             'properties': {'complex_property': {'$ref': '#/components/schemas/RHELVersions'},
  +                                                            'simple_property': {'nullable': True,
  +                                                                                'title': 'Simple '
  +                                                                                         'Property',
  +                                                                                'type': 'boolean'}},
  +                                             'title': 'DeviceEntry',
  +                                             'type': 'object'},
"""

```

### Description

Run pytest {path_tothe_module}. 
Simple property has nullable in the apispec, but complex property doesn't.

### Environment

* OS: [e.g. Linux / Windows / macOS]: Linux
* FastAPI Version [e.g. 0.3.0]: 0.62

To know the FastAPI version use:

```bash
python -c "import fastapi; print(fastapi.__version__)"
```

* Python version: 3.9

To know the Python version use:

```bash
python --version
```

**Comment by Mause** [comment]
I think you may be confusing features in swagger and openapi - openapi uses the required array field to indicate if a field is optional I believe

**Comment by zhukovgreen** [comment]
@Mause thx for following this up. I am not sure I understand you. I adjusted the description. 
Refs: https://github.com/OAI/OpenAPI-Specification/issues/1368

**Comment by Mause** [comment, marked as answer]
See the following example:

```python

import enum

from typing import Optional

from pydantic import BaseModel


class RHELVersions(enum.Enum):
    """Available RHEL versions within the database."""

    el7: str = "RHEL 7"
    el8: str = "RHEL 8"
    el9: str = "RHEL 9"


class DeviceEntry(BaseModel):
    """A model which represents the single hardware device."""

    simple_property1: Optional[bool]
    complex_property1: Optional[RHELVersions]
    simple_property2: bool
    complex_property2: RHELVersions


assert DeviceEntry.schema() == {
    'definitions': {
        'RHELVersions': {
            'description': 'Available RHEL versions ' 'within the database.',
            'enum': ['RHEL 7', 'RHEL 8', 'RHEL 9'],
            'title': 'RHELVersions',
        }
    },
    'description': 'A model which represents the single hardware device.',
    'properties': {
        'complex_property1': {'$ref': '#/definitions/RHELVersions'},
        'complex_property2': {'$ref': '#/definitions/RHELVersions'},
        'simple_property1': {'title': 'Simple Property1', 'type': 'boolean'},
        'simple_property2': {'title': 'Simple Property2', 'type': 'boolean'},
    },
    'required': ['simple_property2', 'complex_property2'],
    'title': 'DeviceEntry',
    'type': 'object',
}
```
Note the `required` field - this how you mark whether a field is optional or not

**Comment by zhukovgreen** [comment]
@Mause thanks for your explanation. Do you know why is there a nullable attribute of the property?

**Comment by Mause** [comment]
There isn't - if you look at the pydantic documentation: https://pydantic-docs.helpmanual.io/usage/schema/#field-customisation

If you look at the end of the function documentation, you just added some fields via the `**extra` which was included verbatim

**Comment by zhukovgreen** [comment]
@Mause ok, thank you for your help

**Comment by tiangolo** [comment]
Thanks for the help here @Mause ! :clap: :bow: 

Thanks for reporting back and closing the issue @ZhukovGreen  :+1:
